### PR TITLE
Use same naming style for modules as classes

### PR DESCRIPTION
**Pull request by Youssef1313**
Related to https://github.com/dotnet/roslyn-analyzers/pull/4086#discussion_r479483657
cc: @mavasani

### CODE REVIEW

**Comment by Youssef1313** [comment]
Marking as a draft as I think this needs additional changes to fix the issue.

**Comment by Youssef1313** [comment]
@mavasani In https://github.com/dotnet/roslyn/blob/master/src/Analyzers/VisualBasic/Tests/NamingStyles/NamingStylesTests.vb, all VB tests contain a PascalCase module. Not sure what's wrong. Also, I'm not sure how to continue with explicitly supporting "Module" in editorconfig.

**Comment by Youssef1313** [comment]
Alright, I've just seen your comment https://github.com/dotnet/roslyn-analyzers/pull/4086#discussion_r479493125.
However, I think Module is still not configurable in .editorconfig without this PR.

**Comment by Youssef1313** [comment]
Expecting the following tests to fail after adding the missing resource:

https://github.com/dotnet/roslyn/blob/163404b1385780bee5fc8a19a9ffc8ca44fdb773/src/VisualStudio/Core/Test/Options/BasicEditorConfigGeneratorTests.vb#L19

https://github.com/dotnet/roslyn/blob/163404b1385780bee5fc8a19a9ffc8ca44fdb773/src/VisualStudio/Core/Test/Options/CSharpEditorConfigGeneratorTests.vb#L18

**Comment by Youssef1313** [comment]
@mavasani Can you give this a look?
This PR is to support a VB Module naming preference. Currently it doesn't seem to be supported:

https://github.com/dotnet/roslyn/blob/7fa4e05dcf9be5472ce9ed00fe41a19c223eb348/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/NamingStyles/EditorConfig/EditorConfigNamingStyleParser_SymbolSpec.cs#L64-L79

Do you want to support it? If yes, how I should move forward with this? If not, you can close.

**Comment by mavasani** [comment]
@Youssef1313 Can you please add tests to show what is missing?

**Comment by Youssef1313** [comment]
@mavasani I'm not sure where the tests should go, but basically what I'm trying to do is support something like:

```ini
dotnet_naming_symbols.module.applicable_kinds = module # VB-specific.
```

I'm not sure whether you currently make modules follow the same rules for classes, or they're just always unchecked.

**Comment by mavasani** [comment]
@Youssef1313 Will you be able to add tests to https://github.com/dotnet/roslyn/blob/master/src/Analyzers/VisualBasic/Tests/NamingStyles/NamingStylesTests.vb?

**Comment by Youssef1313** [comment]
@mavasani This is the scenario I'm trying to fix here:

image

**Comment by Youssef1313** [comment]
From [Kind of symbols](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/naming-rules#kinds-of-symbols) section in docs, modules doesn't seem to be supported.

There are 2 solutions:

1. Support it as a separate symbol, which is what I was trying to do in this PR (but didn't work until now)
2. Make its style the same as whatever set for classes.

**Review by sharwell — CHANGES_REQUESTED** [review]
❓ Why would we not want this covered by `class`?

**Comment by Youssef1313** [comment]
@sharwell Yeah this should be the desired behavior and the fix should be easy enough, but the 1-year-ago me was stuck on such a trivial fix 😄

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/NamingStyles/EditorConfig/EditorConfigNamingStyleParser_SymbolSpec.cs [hunk 1]
```diff
@@ -171,6 +173,7 @@ private static ImmutableArray<SymbolKindOrTypeKind> ParseSymbolKindList(string s
                 {
                     case "class":
                         builder.Add(_class);
+                        builder.Add(_module);
```
Comment (Youssef1313) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/NamingStyles/EditorConfig/EditorConfigNamingStyleParser_SymbolSpec.cs:176]: @sharwell @mavasani Can you take a look please? Thanks!

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/NamingStyles/Serialization/SymbolSpecification.cs [hunk 2]
```diff
@@ -363,10 +363,25 @@ public SymbolKindOrTypeKind(MethodKind methodKind) : this()
...
+                    return true;
+                }
+
+                if (kind1 == CodeAnalysis.TypeKind.Class && kind2 == CodeAnalysis.TypeKind.Module)
+                {
+                    return true;
+                }
+
+                return kind1 == kind2;
+            }
```
Comment (Youssef1313) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/NamingStyles/Serialization/SymbolSpecification.cs:383]: This is a bad change, I think. It resulted from the tests creating `SymbolSpecification`s themselves, and not relying on the parser. Either tests should use the parser, or the following is updated to include modules:

https://github.com/dotnet/roslyn/blob/3e091aba2b46f8574c41f8431dcbeba42274ca86/src/EditorFeatures/DiagnosticsTestUtilities/NamingStyles/NamingStylesTestOptionSets.cs#L129-L136

**Comment by Youssef1313** [comment]
Replaced with https://github.com/dotnet/roslyn/pull/62141

**Pull request closed without merging**